src/agentic/actor_agents.py

```diff
@@ -716,19 +719,39 @@ def __init__(self, actor_ref: ray.actor.ActorHandle, agent_facade: "RayFacadeAge
     async def handle_post(self, prompt: ProcessRequest) -> str:
         self.prompt = prompt
         level = DebugLevel(prompt.debug) if prompt.debug else DebugLevel(DebugLevel.OFF)
-        return self.agent_facade.start_request(prompt.prompt, debug=level)
+        print(f"Received run_id: {prompt.run_id}")
+        return self.agent_facade.start_request(prompt.prompt, debug=level, run_id=prompt.run_id)
 
     @app.get("/getevents", response_model=None)
     async def get_events(self, request_id: str, stream: bool = False) -> list[str]|EventSourceResponse:
         if not stream:
             results = []
             for event in self.agent_facade.get_events(request_id):
-                results.append(str(event))
+                # Create a serializable version of the event
+                event_data = {
```

Comment:
So this is fine for now. But my whole intention with https://github.com/supercog-ai/agent-protocol was to define a set of JSON schemas which exactly define all the events you would get from the API. I think the way this should work is:
  - agent-protocol defines an Open API spec for the "agent API", which includes JSON schemas for all events
  - agentic.events defines Pydantic models which follow the spec. It may be that we can/should just generate this from the OpenAPI spec.
  - One can serialize these events with nothing more than `model_dump` or the default FastAPI serializer and you get exactly what you expect from the API.